gitpuller_wrapper/main.js

```diff
@@ -0,0 +1,141 @@
+// This code will add a "New Notebooks" tab to your jupyter hub environment 
+// and provide you with a text entry widget that you can paste github links into
+// and use nbgitpuller to bring the repository/notebook into your hub environment.
+
+
+define(function(require){
+    var $ = require('jquery');
+    var Jupyter = require('base/js/namespace');
+    // grab giturl parser 
+    var gh = require('./parse-git');
+    // grab html stuff
+    var page = require('./page_info');
+
+    // This is a function from page_info.js that dumps the html for the page
+    var page_text = html_page()
+    
+    function load() {
+
+        if (!Jupyter.notebook_list) return;
+        var base_url = Jupyter.notebook_list.base_url;
+        // page_text really should be an .html file, but I haven't been able to get 
+        // Jupyter to let me import an html file.
+        $(".tab-content").append(page_text);
+
+        // Set up event handling for changes on the text field. 
+        $( ".target").change(function() {
+            // #repo is defined in html_page()
+            Jupyter.repoPull = $('#repo').val();
+
+            // TODO: Probably grab the hub.callysto bit from the Jupyter info 
+            // so this is general to any hub?
+            var url_head = "https://hub.callysto.ca/jupyter/hub/user-redirect/git-pull?";
```

Comment:
This seems to work for my local jupyter instance -- I don't have a hub I can play with extensions on yet, but it'd sure be convenient if just a relative URL would take care of this:

```
-            var url_head = "https://hub.callysto.ca/jupyter/hub/user-redirect/git-pull?";
+            var url_head = "git-pull?";
```

Reply:
Whoops! Sorry just saw this.

I agree, a relative link would be better, and we can probably get that right from the environment. Currently the main issue is with the recent security updates this can no longer give github a poke to ensure there's no typos in user URLs. Once I get that worked out I'll start digging and update this to relative links. That said, in its current state, this is kind of broken in the sense that it works - but it can no longer check if the repo exists or not.